includes/functions.php

```diff
@@ -262,17 +262,22 @@ function getImageName($device, $use_database = true, $dir = 'images/os/')
 function renamehost($id, $new, $source = 'console')
 {
     $host = gethostbyid($id);
+    $host_rrd_dir = Rrd::dirFromHost($host);
+    $is_remote_rrd = false;
 
-    if (! is_dir(Rrd::dirFromHost($new)) && rename(Rrd::dirFromHost($host), Rrd::dirFromHost($new)) === true) {
+    if (! is_dir($host_rrd_dir) && Rrd::checkRrdExists($host_rrd_dir)) { // We're most likely using a remote rrdcached
+        $is_remote_rrd = true;
+    }
+    if ($is_remote_rrd || (! is_dir(Rrd::dirFromHost($new)) && rename(Rrd::dirFromHost($host), Rrd::dirFromHost($new)) === true)) {
         dbUpdate(['hostname' => $new, 'ip' => null], 'devices', 'device_id=?', [$id]);
         log_event("Hostname changed -> $new ($source)", $id, 'system', 3);
 
-        return '';
+        return ['message' => '', 'manual_rrd_rename' => $is_remote_rrd];
```

Comment:
I suggest throwing an exception instead of returning an array.

Reply:
You're right, I just pushed new changes to change that.